src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -93,41 +93,89 @@ internal partial class PackageInstallerService : AbstractDelayStartedService, IP
             _packageSourceProvider = packageSourceProvider;
         }
 
-        public ImmutableArray<PackageSource> GetPackageSources()
+        public async ValueTask<ImmutableArray<PackageSource>?> TryGetPackageSourcesAsync(bool allowSwitchToMainThread, CancellationToken cancellationToken)
         {
-            // Only read from _packageSources once, since OnSourceProviderSourcesChanged could reset it to default at
-            // any time while this method is running.
-            var packageSources = _packageSources;
-            if (packageSources != null)
+            // Only read from _packageSourcesAsync once, since OnSourceProviderSourcesChanged could reset it to default
```

Comment:
This _looks_ like a violation of the comment given it's doing a double-checked lock pattern here. Do we need the extra complexity versus just deleting the outer check? This isn't a high-traffic lock.

Reply:
Removed the double-checked lock